extension/src/lib/chromeSettings.js

```diff
@@ -57,19 +70,119 @@ fluid.defaults("gpii.chrome.settings", {
                 flowManager: "ws://localhost:8081/browserChannel",
                 retryTime: 10
             }
+        },
+        contextMenuPanel: {
+            type: "gpii.chrome.contextMenuPanel",
+            options: {
+                model: "{settings}.model",
+                strings: {
+                    captions: "youtube captions",
+                    inputsLarger: "enhance inputs",
+                    rightClickToSelect: "right-click to select",
+                    selfVoicing: "text-to-speech",
+                    simplifiedUI: "reading mode",
+                    syllabification: "syllables",
+                    tableOfContents: "table of contents"
+                },
+                distributeOptions: {
+                    reset: {
+                        target: "{that reset}.options.invokers.click",
+                        record: {
+                            func: "{settings}.updateSettings",
+                            args: ["{settings}.options.defaultSettings"]
+                        }
+                    },
+                    preferences: {
+                        target: "{that subMenu}.options.components",
+                        record: {
+                            "syllabification": {
+                                type: "gpii.chrome.contextItem.checkbox",
+                                options: {
+                                    contextProps: {
+                                        title: "{contextMenuPanel}.options.strings.syllabification"
+                                    },
+                                    model: {
+                                        value: "{contextMenuPanel}.model.settings.syllabificationEnabled"
+                                    }
+                                }
+                            },
+                            "rightClickToSelect": {
+                                type: "gpii.chrome.contextItem.checkbox",
+                                priority: "after:syllabification",
```

Comment:
These priority annotations are invalid (especially in the new framework) - please remove them

Reply:
@amb26 I left a comment about this in the PR. I wonder if you have any thoughts on how to avoid the. I need these to all be created in a specific order so that they appear in the correct order in the context menu. It appears that they are ordered based on creation.

Reply:
Hi @jobara - it's always going to be problematic to depend on a particular creation order. I recommend you shift any such information into the options proper of the component, either with a simple numerical index, or else put your current priority field value into a standard option named "priority" and then parse and sort records corresponding to the components using fluid.parsePriorityRecords - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/Fluid.js#L1406

Reply:
Some further conversation about this in the [fluid-work channel](http://irc-logs.fluidproject.org/%23fluid-work/%23fluid-work.2019-09-03.log) today suggested that we create all of the sub components in any order, but trigger their calls to the chrome api in the correct sequence.